src/metabase/pulse.clj

```diff
@@ -106,174 +159,215 @@
 (defn- format-cell
   [value col]
   (cond
-    (instance? java.util.Date value) (format-timestamp (.getTime value) col)
+    (instance? Date value) (format-timestamp (.getTime ^Date value) col)
     (and (number? value) (not (datetime-field? col))) (format-number value)
     :else (str value)))
 
 ;;; ## RENDERING
 
-(defn card-href
+(defn- card-href
   [card]
   (h (str (setting/get :-site-url) "/card/" (:id card) "?clone")))
 
-; ported from https://github.com/radkovo/CSSBox/blob/cssbox-4.10/src/main/java/org/fit/cssbox/demo/ImageRenderer.java
-(defn render-to-png
-  [html, os, width]
-  (let [is (new java.io.ByteArrayInputStream (.getBytes html java.nio.charset.StandardCharsets/UTF_8))
-        docSource (new org.fit.cssbox.io.StreamDocumentSource is nil "text/html")
-        parser (new org.fit.cssbox.io.DefaultDOMSource docSource)
-        doc (-> parser .parse)
-        windowSize (new java.awt.Dimension width 1)
-        media (new cz.vutbr.web.css.MediaSpec "screen")]
-    (.setDimensions media (.width windowSize) (.height windowSize))
-    (.setDeviceDimensions media (.width windowSize) (.height windowSize))
-    (let [da (new org.fit.cssbox.css.DOMAnalyzer doc (.getURL docSource))]
-      (.setMediaSpec da media)
-      (.attributesToStyles da)
-      (.addStyleSheet da nil (org.fit.cssbox.css.CSSNorm/stdStyleSheet) org.fit.cssbox.css.DOMAnalyzer$Origin/AGENT)
```

Comment:
`:import`ing Java classes that you use several times usually makes the code easier to read